pkg/keycloak/keycloak.go

```diff
@@ -33,9 +34,15 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to get authenticated client for keycloak")
 	}
+
 	if event.Deleted {
-		return h.deleteKeycloak(kcCopy)
+		return nil
+	}
+
+	if kcCopy.GetDeletionTimestamp() != nil {
+		return h.finalizeKeycloak(kcCopy)
 	}
```

Comment:
Is there a better way to do this? Currently we need to check if this is set in order to run whatever we need to do when the object is deleted. Currently it just removes the finalizer we are setting in order for the object to actually get deleted.

Reply:
I think this shouldn't be done at all, @dimitraz is writing the deprovision code that will remove the finalizer when the deprovision is completed.

Reply:
This is here in this PR because this PR is adding a finalizer, and if we don't deal with removing it as well, you have to manually remove it to get resources to be deleted which is really annoying.

What i was asking anyway, was is there a better way of knowing if this object is marked for deletion, or do you just check if that timestamp has been set to something?

Reply:
yeah, that's fine, it's the same conclusion @dimitraz and @jameelb came to in their work,